fe1-web/src/core/redux/ServerReducer.ts

```diff
@@ -0,0 +1,105 @@
+/**
+ * This error is disabled since reducers use the createSlice function, which requires the user to
+ * param-reassign. Please do not disable other errors.
+ */
+/* eslint-disable no-param-reassign */
+import { createSelector, createSlice, Draft, PayloadAction } from '@reduxjs/toolkit';
+
+import { PublicKey, Server, ServerAddress, ServerState } from 'core/objects';
+
+/**
+ * Reducer & associated function implementation to store all known servers
+ */
+
+export interface ServerReducerState {
+  byAddress: Record<ServerAddress, ServerState>;
+  allAddresses: ServerAddress[];
+}
+
+const initialState: ServerReducerState = {
+  byAddress: {},
+  allAddresses: [],
+};
+
+export const SERVER_REDUCER_PATH = 'servers';
+
+const serverSlice = createSlice({
+  name: SERVER_REDUCER_PATH,
+  initialState,
+  reducers: {
+    addServer: (state: Draft<ServerReducerState>, action: PayloadAction<ServerState>) => {
+      const server = action.payload;
+
+      state.byAddress[server.address] = server;
+      state.allAddresses.push(server.address);
+    },
```

Comment:
If you add another server with the same address, but other properties (by mistake), it will get overwritten, right ? Is that the behaviour you want ?